### ISSUE TITLE
sess.run(fetches)取不出来？

### ISSUE THREAD

**Issue by Nan-Wong98** [zhysora/PSGan-Family#6]
作者您好，我下载了GF-2的数据库，然后按顺序最终生成patch_size=64的tfrecords，因为是windows系统，因此运行的不是run.py，而是psgan.py，对args做了如下改动：
parser.add_argument("--train_tfrecord", help="filename of train_tfrecord",default="D:/file_wn/PSGan-Family- 
                                     master/data/zh/PSData/TFRecords/GF-2_train_64.tfrecords")
parser.add_argument("--test_tfrecord", help="filename of test_tfrecord", default="D:/file_wn/PSGan-Family- 
                                     master/data/zh/PSData/TFRecords/GF-2_origin_test_64.tfrecords")
parser.add_argument("--mode", default="train")
parser.add_argument("--output_dir", default="D:/file_wn/PSGan-Family-master/data/zh/PSOutput/GF-2_train_64_psgan.tfrecords")
parser.add_argument("--checkpoint", default=None, help="directory with checkpoints")
parser.add_argument("--max_steps", type=int, help="max training steps")
parser.add_argument("--max_epochs", type=int, help="number of training epochs",default=50)
parser.add_argument("--summary_freq", type=int, default=0, help="update summaries every summary_freq steps")
parser.add_argument("--progress_freq", type=int, default=200, help="display progress every progress_freq steps")
parser.add_argument("--trace_freq", type=int, default=10, help="trace execution every trace_freq steps")
parser.add_argument("--display_freq", type=int, default=0, help="write current training images ever display_freq steps")
parser.add_argument("--save_freq", type=int, default=1000, help="save model every save_freq steps")

parser.add_argument("--batch_size",type=int, default=8, help="number of images in batch")

parser.add_argument("--lr", type=float, default=0.0001, help="initial learning rate for adam")
parser.add_argument("--beta1", type=float, default=0.5, help="momentum term of adam")
parser.add_argument("--l1_weight", type=float, default=100.0, help="weight on L1 term for generator gradient")
parser.add_argument("--gan_weight", type=float, default=1.0, help="weight on GAN term for generator gradient")

parser.add_argument("--ndf", type=int, default=32, help="number of generator filters in first conv layer")
parser.add_argument("--train_count", type=int, default=20765,help="number of training data")
parser.add_argument("--test_count", type=int, default=960, help="number of test data")
parser.add_argument("--gpus", help="gpus to run", default="1")
parser.add_argument("--blk", type=int, default=64, help="size of sample lr image")

但是，貌似没有跑起来，于是我试着输出fetches，却无法输出，问题应该就出在sess.run(fetches)这里，请问您知道问题出在哪里吗？

**Comment by zhysora** [comment]
有具体的报错吗

**Comment by Nan-Wong98** [comment]
> 有具体的报错吗

没有，就显示一直在运行，但是出不来结果，我只改动了args，我再找找原因吧，谢谢您的回复

**Comment by Nan-Wong98** [comment]
> 有具体的报错吗

作者您好，在一通操作之后，result终于可以输出了，但是显示：
{'train': None, 'global_step': 1}
而train被定义为：tf.group(update_losses, incr_global_step, gen_train)
这是为什么呢？是因为没有feed_dict吗？这是我一直比较疑惑的点

**Comment by bigchang** [comment]
> 有具体的报错吗

作者您好，在一通操作之后，result终于可以输出了，但是显示：
{'train': None, 'global_step': 1}
而train被定义为：tf.group(update_losses, incr_global_step, gen_train)
这是为什么呢？是因为没有feed_dict吗？这是我一直比较疑惑的点

**Comment by zhysora** [comment]
train这么定义依次对应每次更新loss值，更新当前迭代次数，梯度下降。在构建计算图时，在load_example()给create_model()输入后，数据输入流就已经建立了，不需要再feed_dict。
你可以试着看看update_losses, incr_global_step, gen_train这3个里面那个变成None了，循着往前找，看计算图是从哪出的问题

**Comment by Nan-Wong98** [comment]
> train这么定义依次对应每次更新loss值，更新当前迭代次数，梯度下降。在构建计算图时，在load_example()给create_model()输入后，数据输入流就已经建立了，不需要再feed_dict。
> 你可以试着看看update_losses, incr_global_step, gen_train这3个里面那个变成None了，循着往前找，看计算图是从哪出的问题

作者您好，之前的问题已得到解决，原来是我设置cpu在跑，所以输出日志需要更长时间，实际上results一直是有的，只是查看train这个'op'会显示None。目前在psgan上跑了两个，一个是如论文中超参设置的，另一个则是将gan_weight设置为0。但我发现，前者即便加入了gan_loss，且在训练过程中某段区间内与l1_loss显示了一定的相关性，但l1_loss似乎并没有明显的下降。
如图：
gan_weight=0:https://user-images.githubusercontent.com/30307057/114492178-a6824400-9c4a-11eb-9c46-708f778a9cfb.PNG
gan_weight=1:https://user-images.githubusercontent.com/30307057/114492219-bb5ed780-9c4a-11eb-826f-218f05e4eee2.PNG
gan_weight为1时，gan_loss：https://user-images.githubusercontent.com/30307057/114492480-44760e80-9c4b-11eb-9d74-d92260cfbd9b.PNG
所以，如何判断对抗是否在work呢？还是说，不应该仅凭l1_loss去判断生成图像的好坏，而是使用定量评价的结果去判断？

**Comment by zhysora** [comment]
一般可以通过具体的图像结果，从视觉效果上看

**Comment by Nan-Wong98** [comment]
> 一般可以通过具体的图像结果，从视觉效果上看

视觉上的效果感觉已经不太能辨别出来差别了，我在另一个GAN框架上也是这样类似的现象，加入GAN网络貌似并不能使L2loss或L1loss减小，而生成器的生成结果在视觉上都差不多，所以就放弃了进一步做定量评价，我主要关注L1loss是否会因为GAN网络的加入而减小。那视觉上差不多的情形下，是否需要做进一步的定量评价呢？

**Comment by zhysora** [comment]
这个看你的需求了，一般来说引入GAN是为了让图像更加逼真，loss真不一定会减小；而且你说的这个问题确实也是个通病，我对这块理解也不是很深
